p4-14/v1.0.4/tex/p4.pt

```diff
@@ -646,11 +646,11 @@ valid bit for a metadata field, defining a separate metadata flag to
 represent that field's validity is a reasonable work around.
 \end{adjustwidth}
 
-Only valid packet header fields may result in a match (when a value is
-specified for exact or ternary matches against the field), although a
-match operation may explicitly check if a header instance (or field)
-is valid. Only valid packet headers are considered for deparsing (see
-Section~\ref{sec:deparsing}).  
+When an invalid packet header is used as a part of the match key for a table,
+the value of the header is undefined.
```

Comment:
@jafingerhut thank you for the explanation. I would not look forward to debugging such a program, but I can see how this would be (ab)used.